modules/lightning_features/lightning_core/src/BundleEntityStorage.php

```diff
@@ -64,4 +64,23 @@ public function loadMultiple(array $ids = NULL, $check_access = FALSE) {
     return parent::loadMultiple($ids);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function loadByProperties(array $values = [], $check_access = FALSE) {
+    // Build a query to fetch the entity IDs.
+    $entity_query = $this->getQuery();
+    $this->buildPropertyQuery($entity_query, $values);
+    if ($check_access) {
+      $result = array_filter($entity_query->execute(), [
+        $this->accessHandler,
+        'createAccess'
+      ]);
+    }
+    else {
+      $result = $entity_query->execute();
+    }
+    return $result ? parent::loadMultiple($result) : [];
```

Comment:
I'd rather do the if ($check_access) and array_filter() call before returning, rather than two separate calls to $entity_query->execute().